sparkler-api/src/main/java/edu/usc/irds/sparkler/Scorer.java

```diff
@@ -0,0 +1,17 @@
+package edu.usc.irds.sparkler;
+
+import edu.usc.irds.sparkler.model.FetchedData;
+import edu.usc.irds.sparkler.model.Resource;
+
+import java.util.Iterator;
+
+/**
+ * A contract for any {@link ExtensionPoint} that offers functionality to 
+ * determine a score for urls.
+ */
+public interface Scorer extends ExtensionPoint {
+
+    Double score(String extractedText) throws Exception;
```

Comment:
Mathematically, scorer is a function which maps `X --> Y`
`Y`, the value, is pretty sure a `double` number.
However, we have to brainstorm for what `X`, the argument, should be! 
I think `X` can be and should be much broader than extracted text.

For concrete example,  my scorer plugin might score based on the url, domain, extension, inlinks (or parent links), outlinks, heading of the page, etc... 

I suggest changing the argument of scorer from String to `CrawlData` or `FetchedData` (which should have some way to get `Resource` object if needed)

It looks like `CrawlData` and `FetchData` are mistakenly added to 'sparkler-app' instead of 'sparkler-api' (you can move it to sparkler-api's sparkler.model package. Doing so doesn't affect other modules since we preserve package name.

Also, let's check with @sujen1412 to get his thoughts too.

Good starting by the way. Because it is interface we are adding, we need to make it future proof.

Reply:
Completely agree with X being the crawl datum object and not just extracted text. 

> It looks like CrawlData and FetchData are mistakenly added to 'sparkler-app' instead of 'sparkler-api' (you can move it to sparkler-api's sparkler.model package. Doing so doesn't affect other modules since we preserve package name.

I believe this is a bigger change than just the scorer and can be put in as another pull request.
So, in order to make progress with the scoring, can we have this merged and another issue opened to make the change to CrawlData and FetchData or is this a blocker ? 
Thoughts @thammegowda ?

Reply:
we can merge this and then later make the requested changes in a new PR 👍 

I think the dynamic fields mapping issue below https://github.com/USCDataScience/sparkler/pull/133#discussion_r145585339 should be fixed in this PR

Reply:
> I think the dynamic fields mapping issue below #133 (comment) should be fixed in this PR

Yes I agree with that.

Reply:
Yes, I agree with that. I am working on this.